### PR TITLE
Use MetadataV3 for peers custody subnet accounting

### PR DESCRIPTION
**Pull request by Nashatyrev**
## PR Description

- Add handling and notifying of Peer Metadata updates
- Add `MetadataDasPeerCustodyTracker`

### CODE REVIEW

File: networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/peers/MetadataDasPeerCustodyTracker.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    metadata
+        .getOptionalCustodySubnetCount()
+        .ifPresent(
+            subnetCount ->
+                connectedPeerSubnetCount.put(peer.getDiscoveryNodeId(), subnetCount.intValue()));
+  }
+
+  @Override
+  public int getCustodyCountForPeer(UInt256 nodeId) {
+    return connectedPeerSubnetCount.getOrDefault(nodeId, 0);
```
Comment (zilm13) [inline networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/peers/MetadataDasPeerCustodyTracker.java:48]: Maybe it makes sense to default it to custody requirements?

**Review by zilm13 — APPROVED** [review]
LGTM

**Comment by zilm13** [comment]
will fix #89

File: networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/peers/MetadataDasPeerCustodyTracker.java (hunk 1 above)
Comment (Nashatyrev) [reply networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/peers/MetadataDasPeerCustodyTracker.java:48]: We are wrapping the tracker with `DasPeerCustodyCountSupplier.capped()` in the `BeaconChainController`. We can actually get rid of it

**Pull request merged**